### PR TITLE
feat(packaging/windows): specify agent restart when getting failure on Windows

### LINKED ISSUE #5781: Windows: auto-restart service on failure
### Request

Change the static mode/flow mode installer for Windows so the service gets auto-restarted automatically on failure. 

This can be done by calling `sc.exe failure` in the NSIS. 

### Use case

Self-healing the agent on intermittent failures.

### PR DESCRIPTION
**Pull request by hainenber**
#### Which issue(s) this PR fixes
Closes #5781 

#### Notes to the Reviewer
Can you check if the given defaults are "sane" enough? 

#### PR Checklist

- [x] CHANGELOG.md updated

### CODE REVIEW

File: packaging/grafana-agent/windows/install_script.nsis [hunk 1]
```diff
@@ -132,6 +132,8 @@ Function Install
     Pop $0
     nsExec::ExecToLog `sc start "Grafana Agent"`
     Pop $0
+    # Auto-restart agent on failure. Reset failure counter after 60 seconds without failure
+    nsExec::ExecToLog `sc failure "Grafana Agent" reset= 60 actions= restart/5000 reboot= "Grafana Agent has failed. Restarting in 5 seconds"`
```
Comment (rfratto) [inline packaging/grafana-agent/windows/install_script.nsis:136]: Can you add this to the Flow NSIS script too?

**Review by rfratto — COMMENTED** [review]
LGTM, but that's only for the static mode installer right now, I'd like to see the change in the Flow installer too.

(And the settings you provided look reasonable to me 👍 We can always tune further if needed)

File: packaging/grafana-agent/windows/install_script.nsis (hunk 1 above)
Comment (hainenber) [reply packaging/grafana-agent/windows/install_script.nsis:136]: Oh there's another one? Let me dig around. Thanks for the lead!

Comment (rfratto) [reply packaging/grafana-agent/windows/install_script.nsis:136]: Yeah, it's right here :) https://github.com/grafana/agent/blob/main/packaging/grafana-agent-flow/windows/install_script.nsis

Comment (rfratto) [reply packaging/grafana-agent/windows/install_script.nsis:136]: Also, I think this needs a `Pop $0` after it (like in the ExecToLog above) to pop the result off the stack.

Comment (hainenber) [reply packaging/grafana-agent/windows/install_script.nsis:136]: Gotcha.

Comment (hainenber) [reply packaging/grafana-agent/windows/install_script.nsis:136]: `Pop $0` added and similar configs has been added for Flow agent's Windows installer in latest commit. PTAL. Thanks!

File: packaging/grafana-agent-flow/windows/install_script.nsis [hunk 2]
```diff
@@ -95,6 +95,10 @@ Section "install"
   # Start the service.
   nsExec::ExecToLog 'sc start "Grafana Agent Flow"'
   Pop $0
+
+  # Auto-restart agent on failure. Reset failure counter after 60 seconds without failure
+  nsExec::ExecToLog `sc failure "Grafana Agent" reset= 60 actions= restart/5000 reboot= "Grafana Agent has failed. Restarting in 5 seconds"`
```
Comment (rfratto) [inline packaging/grafana-agent-flow/windows/install_script.nsis:100]: The service name is different here :) 

```suggestion
  nsExec::ExecToLog `sc failure "Grafana Agent Flow" reset= 60 actions= restart/5000 reboot= "Grafana Agent Flow has failed. Restarting in 5 seconds"`
```

**Review by rfratto — APPROVED** [review]
Thank you!

**Pull request merged**